deps/chakrashim/src/v8string.cc

```diff
@@ -82,98 +82,57 @@ int String::Utf8Length() const {
   return str.length();
 }
 
-template <class CharType>
-static int WriteRaw(
-    JsValueRef ref, CharType* buffer, int start, int length, int options) {
-  if (length == 0) {
-    // bail out if we are required to write no chars
-    return 0;
-  }
-
-  jsrt::StringUtf8 str;
-  if (str.From(ref) != JsNoError) {
-    // error
-    return 0;
+int String::Write(uint16_t *buffer, int start, int length, int options) const {
+  if (length < 0) {
+    // in case length was not provided we want to copy the whole string
+    length = String::kMaxLength;
   }
 
-  int stringLength = str.length();
-  if (stringLength == 0) {
-    if (!(options & String::NO_NULL_TERMINATION)) {
-      // include the null terminate
-      buffer[0] = L'\0';
+  size_t count = 0;
+  if (JsWriteStringUtf16((JsValueRef)this,
+                         buffer + start, length, &count) == JsNoError) {
+    if (count < (unsigned)length && !(options & String::NO_NULL_TERMINATION)) {
+      buffer[count] = 0;
     }
-    return 0;
   }
+  return count;
+}
 
-  if (start < 0 || start > stringLength) {
-    // illegal bail out
-    return 0;
-  }
-
-  // in case length was not provided we want to copy the whole string
-  int count = length >= 0 ? length : stringLength;
-  count = min(count, stringLength - start);
-
+int String::WriteOneByte(
+    uint8_t* buffer, int start, int length, int options) const {
   if (length < 0) {
-    // If length was not provided, assume enough space to hold content and null
-    // terminator.
-    length = count + 1;
+    // in case length was not provided we want to copy the whole string
+    length = String::kMaxLength;
   }
 
-  jsrt::StringConvert::CopyRaw(str + start, count, buffer, length);
-
-  if (count < length && !(options & String::NO_NULL_TERMINATION)) {
-    // include the null terminate
-    buffer[count++] = L'\0';
+  size_t count = 0;
+  if (JsWriteString((JsValueRef)this,
+    (char*)buffer + start, length, &count) == JsNoError) {
```

Comment:
nvm, re-reading `JsWriteString` API. Can you atleast wrap it? `(char*)(buffer + start)` ?